### PR TITLE
Add Liquibase to Node.js demo service for local development

### PR DESCRIPTION
**Pull request by markysoft**
https://eaflood.atlassian.net/browse/PSD-631

### CODE REVIEW

File: .sequelizerc [hunk 1]
```diff
@@ -2,7 +2,5 @@ const path = require('path')
 
 module.exports = {
   "config": path.resolve('./server/config', 'database-config.js'),
-  "models-path": path.resolve('./server/models'),
-  "seeders-path": path.resolve('./server/seeders'),
-  "migrations-path": path.resolve('./server/migrations')
```
Comment (johnwatson484) [inline .sequelizerc:7]: I'm not sure whether to keep this file.  It enables the sequelize-cli, so for example if you want to create a new model with the cli, it generates the model file and the migration.  Or you can generate empty migrations and seeders.

We're not going to be creating any migrations anymore and I don't think you can create models without migrations.  So unless we add seeders it's not adding any value.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,3 +1,3 @@
-@Library('defra-library@v-8') _
+@Library('defra-library@psd-631-liquibase-migrations') _
```
Comment (johnwatson484) [inline Jenkinsfile]: Can be changed back to `v-8` now

File: README.md [hunk 3]
```diff
@@ -94,9 +94,7 @@ The [package.json](package.json) contains scripts to run the different suites of
 
 Tests that rely on containers to provide databases may be run by passing the suite to be run to the docker-compose command. An example command to run the integration tests is shown below.
 
-`docker-compose -f docker-compose.yaml -f docker-compose.test.yaml run ffc-demo-payment-service sh -c "scripts/wait-for-dependencies scripts/wait-for-dependencies && npm run test:integration"`
-
-Note that the command makes use of the [wait-for-dependencies](scripts/wait-for-dependencies) script to ensure the database is ready for the tests to run, and wraps the command in a `sh -c` for compatibility with `tini` in the Docker container.
+`docker-compose -f docker-compose.yaml -f docker-compose.test.yaml run ffc-demo-payment-service sh -c "pm run test:integration"`
```
Comment (johnwatson484) [inline README.md:97]: Typo

```suggestion
`docker-compose -f docker-compose.yaml -f docker-compose.test.yaml run ffc-demo-payment-service sh -c "npm run test:integration"`
```

File: docker-compose.link.yaml [hunk 4]
```diff
@@ -9,10 +9,7 @@ services:
       context: .
       target: development
     command: >
-      sh -c "
-        scripts/wait-for/wait-for ffc-demo-payment-postgres:5432 --
-        npm run migrate &&
-        npm run start:watch"
+      sh -c "npm run start:watch"
```
Comment (johnwatson484) [inline docker-compose.link.yaml:12]: This is already the command in the Dockerfile so no need for any command override.

File: docker-compose.override.yaml [hunk 5]
```diff
@@ -9,7 +9,7 @@ services:
       context: .
       target: development
     image: ffc-demo-payment-service
-    command: sh -c "scripts/wait-for-dependencies && npm run start:watch"
+    command: sh -c "npm run start:watch"
```
Comment (johnwatson484) [inline docker-compose.override.yaml:12]: This is already the command in the Dockerfile so no need for any command override.

File: README.md [hunk 6]
```diff
@@ -119,6 +117,11 @@ Container images are built using Docker Compose, with the same images used to ru
 
 Use Docker Compose to run service locally.
 
+The service uses [Liquibase](https://www.liquibase.org/) to manage database migrations. To ensure the appropriate migrations have been run the utility script `scripts/start` may be run to execute the migrations, then the application.
+
+Alternatively the steps can be run manually:
+* run migrations
+  * `docker-compose -f docker-compose.migrate.yaml run database-up`
```
Comment (johnwatson484) [inline README.md:124]: If you add the `--rm` flag then the liquibase container is removed when the migration is complete so you don't get a build up of stopped liquibase containers.

```suggestion
  * `docker-compose -f docker-compose.migrate.yaml run --rm database-up`
```

File: .sequelizerc (hunk 1 above)
Comment (markysoft) [reply .sequelizerc:7]: good spot, I've removed the liquibase specific files and the references in the package.json.

**Comment by markysoft** [comment]
Regarding the merge to master, these changes only run migrations against containerised dbs for tests and running locally, not against the dev postgres database.

There is a subsequent card to create PR schemas which will connect to the dev database, so it makes sense to update the release pipeline as part of that work, as it is closely related, and all the logic will be in the shared library.

There are no schema changes so the master build will be OK for this PR. We can either block this card and merge it when the other work is done, or merge it with the knowledge that if there are schema changes or deployments to new databases it will be broken in the short term.

**Pull request merged**